### ISSUE TITLE
sample_vi_fd confuses VLC and Celluloid

### ISSUE THREAD

**Issue by cmirceac** [milkv-duo/cvitek-tdl-sdk-cv180x#3]
Hello,
sample_vi_fd should  not start responding to the connecting clients until it can send an I-frame, currently it sends whatever it has in buffer at moment of connecting and that make VLC and Celluloid video players to abort and disconnect.
Issue present in 1.1.1 and current devel (2024.05.02).
The stream could only be played by ffplayer, part of ffmpeg package that is more tolerant to bad input.
This produces issue #98 in the duo-buildroot-sdk.

**Comment by teknoman117** [comment]
I feel like I'm following you around here a bit, sorry. This issue will also affect https://github.com/milkv-duo/cvitek-tdl-sdk-sg200x (The SDK for the SoC in the Duo 256M and Duo S).

It appears that all of the RTSP connection and client logic is handled inside a closed source library bundled in both repos called libcvi_rtsp.so.

**Comment by cmirceac** [comment]
@teknoman117 True, I've dig trough all the spaghetti and reach the same place :(. Wondering if there's an OSS library that can replace this.
